solana/programs/swap-layer/src/processor/initiate/swap/exact_in.rs

```diff
@@ -105,7 +94,9 @@ pub struct InitiateSwapExactIn<'info> {
     dst_swap_token: Box<Account<'info, token::TokenAccount>>,
 
     /// This account must be verified as the source mint for the swap.
-    #[account(address = staged_custody_token.mint)]
+    ///
+    /// NOTE: This is mutable in case we need to burn residual tokens.
```

Comment:
Are the tokens ever burned? The residuals seems to go back to staged_custody_token.

Reply:
Woops, good catch. I removed mut here: https://github.com/wormholelabs-xyz/swap-layer/pull/62/commits/ec6cd4d84cf7805d7e760b683a3359ce149af15d#diff-6ab032298731fedd8da9185f721035324708e2ef941246502accff9f5cded932L97-L99.

I accidentally left this here when I was trying to figure out what to do with the dust after a Jupiter swap.

And now that this is worked out, there is a path to perform an "Exact Out" the same way... something to tackle later.